# -*- coding: utf-8 -*-
# Generated by Django 1.10.2 on 2016-10-05 18:01
from __future__ import unicode_literals

from django.db import migrations
import csv

def add_rater_and_movie_to_data(apps, schema_editor):
    Rater = apps.get_model("movie_app", "Rater")
    Movie = apps.get_model("movie_app", "Movie")
    Rating = apps.get_model("movie_app", "Rating")

    with open('person.csv') as openfile:
        reader = csv.reader(openfile, delimiter = "|")
        for row in reader:
            Rater.objects.create(id=row[0], age=row[1], gender=row[2], occupation=row[3], zipcode=row[4])
    # raise Exception ("BOOM!")

    with open('movie.csv') as openfile:
        reader = csv.reader(openfile, delimiter = "|")
        for row in reader:
            Movie.objects.create(id=row[0], title=row[1], release_date=row[2], video_release_date=row[3], imdb_url=row[4],
                                unknown=row[5], action_genre=row[6], adventure=row[7], animation=row[8], children=row[9],
                                comedy=row[10], crime=row[11], documentary=row[12], drama=row[13], fantasy=row[14], film_noir=row[15],
                                horror=row[16], musical=row[17], mystery=row[18], romance=row[19], sci_fi=row[20], thriller=row[21],
                                war=row[22], western=row[23])
    # raise Exception ("BOOM!")

    with open('rating.csv') as openfile:
        reader = csv.reader(openfile, delimiter = "\t")
        for row in reader:
            user_id = Rater.objects.get(id=row[0])
            movie_id = Movie.objects.get(id=row[1])
            Rating.objects.create(user_id=user_id, movie_id=movie_id, rating=row[2], time_stamp=row[3])
    # raise Exception ("BOOM!")

class Migration(migrations.Migration):

    dependencies = [
        ('movie_app', '0003_rating'),
    ]

    operations = [
        migrations.RunPython(add_rater_and_movie_to_data)
    ]
